src/extensions/extensionsHelper.ts

```diff
@@ -479,51 +568,78 @@ export async function incrementPrereleaseVersion(
 }
 
 /**
- * Validates the Extension spec.
+ * Gets a list of the next version to upload by release stage.
  *
- * @param publisherId the ID of the Publisher
- * @param extensionId the ID of the Extension
- * @param rootDirectory the directory with the Extension's source
- * @param latestVersion the latest version of the Extension (if any)
- * @param stage the release stage
+ * @param extensionRef the ref of the extension
+ * @param version the new version of the extension
+ */
+async function getNextVersionByStage(
+  extensionRef: string,
+  newVersion: string
+): Promise<{ versionByStage: Map<string, string>; hasVersions: boolean }> {
+  const versionByStage = new Map(
+    ["rc", "alpha", "beta"].map((stage) => [
+      stage,
+      semver.inc(`${newVersion}-${stage}`, "prerelease", undefined, stage)!,
+    ])
+  );
+  let extensionVersions: ExtensionVersion[] = [];
+  try {
+    extensionVersions = await listExtensionVersions(extensionRef, `id="${newVersion}"`, true);
+    for (const extensionVersion of extensionVersions) {
+      const version = semver.parse(extensionVersion.spec.version)!;
+      // Extensions only support semvers of format: <major>.<minor>.<patch>-<stage>.<count>
+      const stage = String(version.prerelease[0]).split(".")[0];
```

Comment:
This will break if version.prerelease is length 0.
```suggestion
      const stage = String(version.prerelease.?[0]).split(".")[0];
```